de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/handlers/FileImportHandler.java

```diff
@@ -1,93 +1,71 @@
 import org.eclipse.e4.core.di.annotations.Execute;
-import org.eclipse.e4.ui.services.IServiceConstants;
-import org.eclipse.jface.resource.ImageDescriptor;
-import org.eclipse.swt.graphics.Point;
-import org.eclipse.swt.widgets.Shell;
-
 import de.tu_bs.cs.isf.e4cf.core.file_structure.FileTreeElement;
 import de.tu_bs.cs.isf.e4cf.core.file_structure.WorkspaceFileSystem;
 import de.tu_bs.cs.isf.e4cf.core.file_structure.util.FileHandlingUtility;
-import de.tu_bs.cs.isf.e4cf.core.util.RCPMessageProvider;
-import de.tu_bs.cs.isf.e4cf.core.util.services.RCPDialogService;
-import de.tu_bs.cs.isf.e4cf.core.util.services.RCPImageService;
 import de.tu_bs.cs.isf.e4cf.core.util.services.RCPSelectionService;
-import de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards.FileImportWizard;
-
+import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
 
 /**
- * A handler that copies files into a selected directory.
+ * A handler that opens the systems file import dialog.
  * 
- * @author Oliver Urbaniak
  */
 public class FileImportHandler {
-	private static final String WINDOW_TITLE = "Import Files";
-	private static final Point WINDOW_SIZE = new Point(800, 600);
-	private static final String FILE_ICON_PATH = "icons/Explorer_View/items/file32.png";
-	
+
 	@Execute
-	public void execute(RCPImageService imageService, RCPDialogService dialogService, RCPSelectionService selectionService,
-			@Named(IServiceConstants.ACTIVE_SHELL) Shell shell, WorkspaceFileSystem fileSystem) {
-		try {
-			Path target = getTargetPath(selectionService);
-			FileImportWizard wizard = buildFileImportWizard(shell, imageService);
-			dialogService.constructDialog(WINDOW_TITLE, WINDOW_SIZE, wizard).open();
-			copyFiles(target, wizard.getSourceFiles(), fileSystem);
-		} catch (IOException e) {
-			RCPMessageProvider.errorMessage(WINDOW_TITLE, e.getMessage());
-		} catch (NullPointerException e ) {
-			RCPMessageProvider.errorMessage(WINDOW_TITLE, e.getMessage());
+	public void execute(RCPSelectionService selectionService, WorkspaceFileSystem workspaceFileSystem) {
+		Path target = Paths.get(workspaceFileSystem.getWorkspaceDirectory().getAbsolutePath());
+
+		final FileChooser fileChooser = new FileChooser();
+		fileChooser.getExtensionFilters().addAll(new ExtensionFilter("All Files", "*.*"));
+		List<File> selectedFiles = fileChooser.showOpenMultipleDialog(null);
+
+		if (selectedFiles != null && !selectedFiles.isEmpty()) {
+			try {
+				target = getTargetPath(selectionService);
+				for (File file : selectedFiles) {
+					workspaceFileSystem.copy(file.toPath(), target);
+				}
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
 		}
-		
 	}
 
-	private void copyFiles(Path target, List<Path> selectedFiles, WorkspaceFileSystem fileSystem) throws IOException {
-		for (Path selectedFile : selectedFiles) {
-			fileSystem.copy(selectedFile, target);
+	@CanExecute
+	public boolean canExecute(RCPSelectionService selectionService) {
+		try {
```

Comment:
this way of determining if a selection is valid seems nuts to me.
Why bother creating custom exceptions that are swallowed anyways?

Reply:
Agreed, let's get rid of the exceptions. Maybe `private boolean isValidSelection(element)`?

Reply:
was corrected and simplified with the general change to handlers